base/linalg/uniformscaling.jl

```diff
@@ -127,8 +127,8 @@ inv(J::UniformScaling) = UniformScaling(inv(J.λ))
 *(J1::UniformScaling, J2::UniformScaling) = UniformScaling(J1.λ*J2.λ)
 *(B::BitArray{2}, J::UniformScaling) = *(Array(B), J::UniformScaling)
 *(J::UniformScaling, B::BitArray{2}) = *(J::UniformScaling, Array(B))
-*(A::AbstractMatrix, J::UniformScaling) = J.λ == 1 ? A : J.λ*A
-*(J::UniformScaling, A::AbstractVecOrMat) = J.λ == 1 ? A : J.λ*A
+*(A::AbstractMatrix, J::UniformScaling) = J.λ == 1 ? copy(A) : J.λ*A
+*(J::UniformScaling, A::AbstractVecOrMat) = J.λ == 1 ? copy(A) : J.λ*A
```

Comment:
Neither the new definition nor the previous one is type-stable (eg. `UniformScaling(1.0) * eye(Int, 3)` returns a `Matrix{Int}`). Should it be fixed?